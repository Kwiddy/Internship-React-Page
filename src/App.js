import React from "react";
import logo from "./logo.svg";
import "./App.css";

function App() {
  return (
    <div>
      <div className="App">Herllsdfo, World!</div>

      <link
        rel="stylesheet"
        href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
      />
      <link
        rel="stylesheet"
        href="https://fonts.googleapis.com/icon?family=Material+Icons"
      />
    </div>
  );
}

export default App;
