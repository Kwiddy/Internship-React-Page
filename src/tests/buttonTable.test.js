import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { render, wait } from "@testing-library/react";
import { shallow, configure, mount } from "enzyme";
import renderer from "react-test-renderer";
import Button from "@material-ui/core/Button";
import DetTable from "../components/accountDetails.js";
import IconButton from "@material-ui/core/IconButton";
import sinon, { mock } from "sinon";
import App from "../App";
import * as f from "../App";
import ButtonTable from "../components/buttonTable.js";

configure({ adapter: new Adapter() });

describe("ButtonTable", () => {
  const mockFunction = jest.fn();
  const wrapper = shallow(<ButtonTable onClickHandler={mockFunction} />);

  const useStateSpy = jest.spyOn(React, "useState");
  const setState = jest.fn();
  useStateSpy.mockImplementation((init) => [init, setState]);

  it("updateButton should trigger onClick", () => {
    wrapper.find("#updateButton").simulate("click");
    expect(mockFunction).toHaveBeenCalled();
    expect(mockFunction.mock.calls[0][0]).toBe("Update");
    // const alertComp = ReactTestUtils.renderIntoDocument(<Alert />);
    // expect(ReactTestUtils.isDOMComponent(<Alert />)).toBe(true);

    // Expect a change of state on button click
    expect(setState).toBeTruthy();

    // const wrapper2 = shallow(<App />);
    // expect(wrapper2.instance().state.open).toBe(true);
  });
  it("cancelButton should trigger onClick", () => {
    wrapper.find("#cancelButton").simulate("click");
    expect(mockFunction).toHaveBeenCalled();
    expect(mockFunction.mock.calls[1][0]).toBe("Cancel");
  });
});

// describe("Spy test", () => {
//   // const component = shallow(<App />);
//   // const compSpy = jest.spyOn(component.instance(), "updateServices");
//   // expect(compSpy).toHaveBeenCalledTimes(1);
//   it("clicks it", () => {
//     let app = shallow(<App />);
//     console.warn(app);
//     const spy = jest.spyOn(app.instance(), "updateServices");

//     // instance.forceUpdate();

//     // const p = app.find('.App-intro')
//     // p.simulate('click')
//     // expect(spy).toHaveBeenCalled()
//   });
// });
