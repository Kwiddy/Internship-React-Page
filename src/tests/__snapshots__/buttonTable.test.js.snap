// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTable Should render the button table 1`] = `ShallowWrapper {}`;
